woudc_data_registry/models.py

```diff
@@ -904,6 +904,94 @@ def __repr__(self):
         return 'DataRecord({}, {})'.format(self.data_record_id, self.url)
 
 
+class Contribution():
+    """
+    Data Registry Contribution
+
+    """
+    __tablename__ = 'contributions'
+
+    id_field = 'contribution_id'
+    id_dependencies = ['project_id', 'dataset_id', 'station_id',
+                       'instrument_name']
+
+    project_id = Column(String, ForeignKey('projects.project_id'),
+                        nullable=False)
+    contribution_id = Column(String, primary_key=True)
+    dataset_id = Column(String, ForeignKey('datasets.dataset_id'),
+                        nullable=False)
+    station_id = Column(String, ForeignKey('stations.station_id'),
+                        nullable=False)
+    station_name = Column(String, nullable=False)
+    country_id = Column(String, nullable=False)
+    country_name_en = Column(String, nullable=False)
+    country_name_fr = Column(String, nullable=False)
+    instrument_id = Column(String, nullable=False)
+    instrument_name = Column(String,
+                             ForeignKey('instruments.name'), nullable=False)
+    instrument_model = Column(String, nullable=False)
```

Comment:
Note that all of these variables that are outside of class methods are _not_ attributes, this is all purely SQL stuff. Because there are joins in SQL the `station_name` and `country_name` and `instrument_name`/`model` are redundant, which is something we want to avoid in the SQL registry.

To make use of joins in the Python code, see other model classes and how they use `relationship`.
For example, instead of having a separate instrument name you could do `instrument = relationship('Instrument', backref=__tablename__)` and then have access to `instrument.name` and `instrument.model` through that.

See https://docs.sqlalchemy.org/en/13/orm/backref.html or look for more in the sqlalchemy docs.